scene/resources/animation_track_filter.h

```diff
@@ -0,0 +1,64 @@
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef ANIMATION_TRACK_FILTER_H
+#define ANIMATION_TRACK_FILTER_H
+
+#include "core/io/resource.h"
+#include "core/variant/typed_array.h"
+
+class AnimationTrackFilter : public Resource {
+	GDCLASS(AnimationTrackFilter, Resource);
+
+	StringName preset_name;
+	Vector<NodePath> tracks;
+
+protected:
+	static void _bind_methods();
+
+public:
+	StringName get_preset_name() const { return preset_name; }
+	void set_preset_name(const StringName p_preset_name) { preset_name = String(p_preset_name).strip_edges(); }
+
+	Vector<NodePath> get_tracks() const { return tracks; }
+	void set_tracks(const Vector<NodePath> &p_tracks) { tracks = p_tracks; }
+
+	TypedArray<NodePath> get_tracks_bind() const;
+	void set_tracks_bind(const TypedArray<NodePath> &p_tracks);
+
+	bool has_track(const NodePath &p_track_path) const;
+	void remove_track(const NodePath &p_track_path);
+	void add_track(const NodePath &p_track_path);
+	void clear_tracks();
+
+	Ref<AnimationTrackFilter> duplicate_without_preset_name() const;
+};
+
+#endif // ANIMATION_TRACK_FILTER_H
```

Comment:
```suggestion
#endif // ANIMATION_TRACK_FILTER_H

```

Reply:
Another issue is it seem git can't recognize the change of end with a new line. I already add a new line at local files and push them.

Reply:
Indeed. I remember getting a notice like "No new line" if there is no line, but if there is a new line, we may not get anything...